microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,242 @@
+        tx_pps = testpmd.get_tx_pps_from_testpmd_output(testpmd_output)
+        log.info(
+            f"TX-PPS:{tx_pps} from {test_nic._upper}/{test_nic._lower}:"
+            + f"{test_nic._pci_slot}"
+        )
+        assert_that(tx_pps).described_as(
+            f"TX-PPS ({tx_pps}) should have been greater than 2^20 (~1m) PPS."
+        ).is_greater_than(2 ** 20)
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        result = node.execute(cmd, sudo=True, shell=True)
+        result.assert_exit_code()
+        return result.stdout
+
+    def _hugepages_init(self, node: Node) -> None:
+        mount = node.tools[Mount]
+        mount.mount("nodev", "/mnt/huge", type="hugetlbfs")
+        mount.mount("nodev", "/mnt/huge-1G", type="hugetlbfs", options="'pagesize=1G'")
+
+    def _hugepages_enable(self, node: Node, log: Logger) -> None:
+        self._execute_expect_zero(
+            node,
+            (
+                "sh -c 'echo 4096 > "
+                "/sys/devices/system/node/node0/hugepages/"
+                "hugepages-2048kB/nr_hugepages'"
+            ),
+        )
+        self._execute_expect_zero(
+            node,
+            (
+                "sh -c 'echo 1 > "
+                "/sys/devices/system/node/node0/hugepages/"
+                "hugepages-1048576kB/nr_hugepages'"
+            ),
+        )
+        node.execute("grep -i huge /proc/meminfo && ls /mnt/", shell=True)
+
+
+class NicInfo:
+
+    # Class for info about an single upper/lower nic pair.
+    # devices using SRIOV on azure typically have an upper synthetic device
+    # paired with a lower SRIOV Virtual Function (VF) device that
+    # enables the passthrough to the physical NIC.
+
+    def __init__(self, upper: str, lower: str = "", pci_slot: str = "") -> None:
+        self._has_secondary = lower != "" or pci_slot == ""
+        self._upper = upper
+        self._lower = lower
+        self._pci_slot = pci_slot
+
+    def has_secondary(self) -> bool:
+        return self._has_secondary
+
+    def __str__(self) -> str:
+        return (
+            "NicInfo:\n"
+            f"upper: {self._upper}\n"
+            f"lower: {self._lower}\n"
+            f"pci_slot: {self._pci_slot}\n"
+        )
+
+    def testpmd_include(self, vdev_type: str) -> str:
+        assert_that(self._has_secondary).is_true().described_as(
+            (
+                f"This interface {self._upper} does not have a lower interface "
+                "and pci slot associated with it. Aborting."
+            )
+        )
+        return f'--vdev="{vdev_type},iface={self._upper}" --allow "{self._pci_slot}"'
+
+
+class NodeNicInfo:
+
+    # Class for all of the nics on a node. Contains multiple NodeNic classes.
+    # Init identifies upper/lower paired devices and the pci slot info for the lower.
+
+    _pci_slot_regex = r"([0-9a-fA-F]{1-4}:)+[0-9a-fA-F]{1-4}"
+    _nics: Dict[str, NicInfo] = dict()
```

Comment:
Don't declare variables in class level in Python, because it would be a class variable. It means all objects share the same variable. Declare it in `__int__`.

Reply:
fixed